[master] Various small fixes and enhacements for some tests

meaksh opened the pull request:
### What does this PR do?

This PR simply introduces a few enhacements and fixes to some of the tests so they behave better when running them on certain environments, i.a. inside GH action containers, or in some particular environmets.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

agraul committed:
Fix mock order in test_dig (test_network.py)

`salt.utils.path.which` should not be mocked before `network.__utils__`. The
Salt loader calls `salt.utils.network.linux_interfaces`, which needs the real
`salt.utils.path.which`.

meaksh committed:
Fix some failing test causing problem in SUSE environments

meaksh committed:
Skip some tests that does not run well in GH Actions

meaksh committed:
Fix formatting and linting issues detected by pre-commit

s0undt3ch commented on tests/pytests/integration/daemons/test_memory_leak.py:
```diff
@@ -46,6 +49,7 @@ def file_add_delete_sls(testfile_path, base_env_state_tree_root_dir):
 
 @pytest.mark.skip_on_fips_enabled_platform
 @pytest.mark.skip_on_darwin(reason="MacOS is a spawning platform, won't work")
+@pytest.mark.skipif(GITHUB_ACTIONS, reason="Test is failing in GitHub Actions")
```
```suggestion
@pytest.mark.skip_on_env("GITHUB_ACTIONS", present=True)
```

s0undt3ch replied:
However, this test is not failing on our test suite...

meaksh replied:
It seems this one is flaky and fails randomly. It feels that this happens more often when running on GH.

s0undt3ch commented on tests/pytests/scenarios/failover/multimaster/test_failover_master.py:
```diff
@@ -12,7 +12,10 @@
+GITHUB_ACTIONS = bool(os.getenv("GITHUB_ACTIONS"))
 
+
+@pytest.mark.skipif(GITHUB_ACTIONS, reason="Test is failing in GitHub Actions")
```
Same as previous file

meaksh committed:
test_pkg: Adjust package expectation for SUSE family

meaksh committed:
Fix tests failing due wrong docker-py version

test_version: skip test in packaged scenario when setup.py is missing

Fix issue related to docker version used during testing

meaksh committed:
Fix test errors when setup.py is not available

test_loader: do not run if setup.py is missing

test_install: Fix test errors when setup.py is not available

meaksh committed:
test_master: use a right service name expected on SUSE family

meaksh committed:
test_jinja_filters: prevent test failure when which binary is not available

meaksh committed:
Prevent errors when x509 utils cannot be loaded

meaksh committed:
test_thin: skip test if virtualenv binary is missing

meaksh committed:
Fix formatting and lint issues after rebase

meaksh committed:
Use pytest.mark.skip_on_env to have a cleaner code

meaksh committed:
Remove unused import to fix pylint issue

meaksh commented:
@dwoz any chance to review this one? TIA!

meaksh committed:
Do not skip test_failover_master::test_pki when running on GH actions

meaksh commented:
> Please rebase and fix conflicts. Also, you're skipping a few tests because they don't work in GitHub runners. Could you verify that it is still the case as we are using GitHub runners exclusively now.

Hey @twangboy, thanks for reviewing here. I just rebased the PR and also removed the skip of tests when running on GitHub actions, as I see they are passing now.

meaksh commented on tests/pytests/functional/modules/state/test_jinja_filters.py:
```diff
@@ -987,9 +987,9 @@ def _filter_id(value):
         ),
         Filter(
             name="which",
```
Actually, the `which` is still the valid name for the filter, it is just that we are not making this test to rely on the existance of `which` binary (which might not be available on the testing VMs), to avoid getting a failure if that's the case.
